lib/http2/hpack.c

```diff
@@ -41,6 +41,8 @@ struct st_h2o_hpack_static_table_entry_t {
 struct st_h2o_hpack_header_table_entry_t {
     h2o_iovec_t *name;
     h2o_iovec_t *value;
+    int decode_return_value; /* record the soft error for bad chars in headers */
```

Comment:
I think we do not need to retain the value, since if a soft error has occurred can be found by checking the value of `err_desc`, and in such case the error code would always be `H2O_HTTP2_ERROR_INVALID_HEADER_CHAR`.